pymongo/pool.py

```diff
@@ -231,6 +232,108 @@ def _set_keepalive_times(sock):
     )
 
 
+def _is_lambda() -> bool:
+    if os.getenv("AWS_LAMBDA_RUNTIME_API"):
+        return True
+    env = os.getenv("AWS_EXECUTION_ENV")
+    if env:
+        return env.startswith("AWS_Lambda_")
```

Comment:
This change was needed to fix the issue described in https://jira.mongodb.org/browse/DRIVERS-2623